src/StoreApi/Utilities/RateLimits.php

```diff
@@ -0,0 +1,125 @@
+<?php
+namespace Automattic\WooCommerce\StoreApi\Utilities;
+
+use WC_Rate_Limiter;
+use WC_Cache_Helper;
+
+/**
+ * RateLimits class.
+ */
+class RateLimits extends WC_Rate_Limiter {
+	/**
+	 * Gets a cache prefix.
+	 *
+	 * @param string $action_id Identifier of the action.
+	 * @return string
+	 */
+	protected static function get_cache_key( $action_id ) {
+		return WC_Cache_Helper::get_cache_prefix( 'store_api_rate_limit' . $action_id );
+	}
+
+	/**
+	 * Get current rate limit row from DB and normalize types. This query is not cached.
+	 *
+	 * @param string $action_id Identifier of the action.
+	 * @return object|null Object containing reset and remaining.
+	 */
+	protected static function get_rate_limit_row( $action_id ) {
+		global $wpdb;
+
+		$row = $wpdb->get_row(
+			$wpdb->prepare(
+				"
+					SELECT rate_limit_expiry as reset, rate_limit_remaining as remaining
+					FROM {$wpdb->prefix}wc_rate_limits
+					WHERE rate_limit_key = %s
```

Comment:
we need to add another clause for `rate_limit_expiry > now()`? otherwise `$wpdb` may return expired rows even if there are is a valid rate-limiting row present.